### PR TITLE
xbmgmt flash windows port

### PR DESCRIPTION
**Pull request by AShivangi**

### CODE REVIEW

File: src/runtime_src/core/tools/xbmgmt2/SubCmdFlash.cpp [hunk 1]
```diff
@@ -46,8 +47,83 @@ namespace {
...
+  std::cout << "\tMax power level:\t\t" << info.mMaxPower << "\n";
+  std::cout << "\tMAC address0:\t\t" << info.mMacAddr0 << "\n";
+  std::cout << "\tMAC address1:\t\t" << info.mMacAddr1 << "\n";
+  std::cout << "\tMAC address2:\t\t" << info.mMacAddr2 << "\n";
+  std::cout << "\tMAC address3:\t\t" << info.mMacAddr3 << "\n";
+
+}
+
+// Update shell on the board.
+static void update_shell(unsigned index, std::string flashType,
```
Comment (stsoe) [inline src/runtime_src/core/tools/xbmgmt2/SubCmdFlash.cpp]: unsigned int, const std::string& flashType
Personally I don't like "unsigned" alone. unsigned what?

Comment (AShivangi) [reply src/runtime_src/core/tools/xbmgmt2/SubCmdFlash.cpp]: Thanks for pointing it out. I fixed this is most of the places, might have missed a couple.

**Comment by AShivangi** [comment]
Thanks for the review @rozumx! I'll keep those points in mind and make the changes in the next iteration

File: src/runtime_src/core/tools/xbmgmt2/flash/flasher.h [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+ * under the License.
+ */
+#ifndef FLASHER_H
+#define FLASHER_H
+
+#include "xspi.h"
+#include "xmc.h"
+#include "firmware_image.h"
+#include "xclfeatures.h"
+#include <sys/stat.h>
```
Comment (stsoe) [inline src/runtime_src/core/tools/xbmgmt2/flash/flasher.h]: How does this compile on windows?

File: src/runtime_src/core/tools/xbmgmt2/flash/flasher.cpp [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+#include "flasher.h"
+#include <stddef.h>
+#include <cassert>
+#include <vector>
+#include <string.h>
```
Comment (stsoe) [inline src/runtime_src/core/tools/xbmgmt2/flash/flasher.cpp]: cstring

File: src/runtime_src/core/tools/xbmgmt2/flash/flasher.h (hunk 2 above)
Comment (AShivangi) [reply src/runtime_src/core/tools/xbmgmt2/flash/flasher.h]: Not sure... but it does. Let me clean up the imports before you merge

**Pull request merged**